### ISSUE TITLE
[Feature] Dropout in attention computation

### ISSUE THREAD

**Issue by billziss-gh** [ml-explore/mlx#2732]
Implementations of attention in other frameworks include a dropout after the softmax operation. For example, see Pytorch's documentation for [`MultiheadAttention`](https://docs.pytorch.org/docs/stable/generated/torch.nn.MultiheadAttention.html) and [`scaled_dot_product_attention`](https://docs.pytorch.org/docs/stable/generated/torch.nn.functional.scaled_dot_product_attention.html).

This does not appear to be the case in MLX's [`MultiHeadAttention`](https://github.com/ml-explore/mlx/blob/v0.29.3/python/mlx/nn/layers/transformer.py#L15) and [`scaled_dot_product_attention`](https://github.com/ml-explore/mlx/blob/v0.29.3/mlx/fast.cpp#L575).

Is this intentional or an oversight?

**Comment by angeloskath** [comment]
It is less and less used lately so we didn't add it. I do not think there is any point in adding it to `MultiHeadAttention` in `mlx.nn`. If anybody needs it they can probably add it to their implementation of attention layer.

**Comment by billziss-gh** [comment]
Thanks for your answer.

My exploration of Pytorch's code shows that they weren't always consistent in performing the dropout in all their backends either (at least historically, not sure about their latest version).

If the answer is "we will not be adding it" that is fine with me and we can close the issue. I would recommend a note in the documentation as it is somewhat surprising to find the attention dropout missing for people coming from different frameworks.
